### DISCUSSION TITLE
Ideas for the new design

### DISCUSSION THREAD

**Discussion by ocean90** [GlotPress/GlotPress-WP#1330] [Ideas]
Use this issue to dump your raw ideas/mockups/sketches/…. Don't get too technical, instead, open a new issue.

Use one comment per idea and if you like somebody else's idea, give it a :+1: reaction.

---
- Current Design: https://github.com/GlotPress/GlotPress-WP/wiki/9.-The-New-Design (List of screens and requirements for each.)
- User Roles and Flows: https://docs.google.com/document/d/1CZrjkBmSYtkGAi6DS1igjNs3K2K6SnFDkoxwI19wEpI/edit

---

Logs of the first chat: https://wordpress.slack.com/archives/glotpress/p1457985681000105
Attendees: @chantalcoolsma, @samuelsidler, @ocean90

**Comment by chantalcoolsma** [comment]
Something what is much wanted by Translation Editors, is the ability to give a reason when rejecting a string. The translator should get a notification, so he/she can correct the mistake. Below an example in the current design:

![schermafbeelding 2016-03-11 om 08 58 45](https://cloud.githubusercontent.com/assets/867632/13696599/b8caaece-e767-11e5-9bee-f4c24b56d4d4.png)

**Comment by chantalcoolsma** [comment]
Crowdin has the option to add a comment to a translated string and optionally mark it as an issue. This could ping the Translation Editors with the permalink to the specific string and take a closer look.

![schermafbeelding 2016-03-11 om 09 03 15](https://cloud.githubusercontent.com/assets/867632/13696695/6647cc12-e768-11e5-9ba3-03860c6195e3.png)

**Comment by isaackeyet** [comment]
It would be great to have reject reasons. However, I think adding standard "reject reasons" is the way to go. I used a translation tool for validators only before, and this was a very useful feature. Sometimes you add a comment anyway, but selecting the core reason for the rejection should be sufficient for most users.

The workflow for a validator could then be:
- Approve current string (CMD+Enter) (moves to next string)
- Reject current string (CMD+Delete) (opens Rejection Details)
- Select Language Style Guide (CMD+1)
- Add comment (textarea already selected): "We don't capitalize actions in Swedish"
- Submit (CMD+Enter) (moves to next string)

We may have to use CMD + Shift + Hotkey or similar, but the idea is the same.

Reject reasons could be:
1. Style Guide
2. Glossary
3. Grammar
4. Punctuation
5. Branding
6. Typos
7. Other

A day in the life of a validator:

> ⌘Enter, ⌘Enter, ⌘Enter, ⌘Delete, ⌘6, "'Registrering is misspelled", ⌘Enter, ⌘Enter, ⌘Delete, ⌘4, ⌘Enter, ⌘Delete, ⌘4, ⌘Enter, ⌘Delete, ⌘4, "Please be careful about proper punctuation.", ⌘Enter, ⌘Enter, ⌘Enter, ⌘W

Here's a rough sketch of said UX:
![reject-reason-sketch](https://cloud.githubusercontent.com/assets/3073557/13698169/0749f25c-e7a5-11e5-8078-0a836a788584.png)

**Comment by chantalcoolsma** [comment]
Great idea. I have another one:

I like the way Github shows actions. It would be great if you could click a string, then it collapses open and you would see something similar like this:

![schermafbeelding 2016-03-11 om 10 31 40](https://cloud.githubusercontent.com/assets/867632/13698489/af4beaf4-e774-11e5-9829-6c2891eb5217.png)

For example if a translator translated the string and when an editor approved or rejected the string. If rejected, you could show the standard reject reason.

**Comment by isaackeyet** [comment]
>  It would be great if you could click a string, then it collapses open and you would see something similar like this:

This would be great to have. It would require the ability to store the history of a string indefinitely. As long as this is technically feasible I think we should pursue it.

**Comment by ocean90** [comment]
Let's not get too technical in this issue. :)

> It would require the ability to store the history of a string indefinitely.

There is #209 which is about implementing a framework to attach data to a translation. Data can be status changes, comments, reject reasons or any other type of notes.

**Comment by ocean90** [comment]
Exploring Design for GlotPress: https://glotpress.blog/2016/06/08/exploring-design-for-glotpress/

**Comment by tobifjellner** [comment]
Another thing that would be really useful is highlighting of what changed in the original when a string was marked fuzzy. Best with some kind of text background. Say red background and strikethrough for deleted characters, and green background and bold for new chars. 

I tried to simulate the idea via markdown. Didn't really work. I'll do it differently.

**Comment by ocean90** [comment]
@tobifjellner That's more of a request for a missing functionality, not directly related to design. I think #427 is related to your idea.

**Comment by Presskopp** [comment]
We did another mockup, in connection to give feedback when rejecting, please see: https://github.com/GlotPress/GlotPress-WP/issues/324
